src/ds++/DracoMath.hh

```diff
@@ -229,6 +229,62 @@ constexpr inline double linear_interpolate(double const x1, double const x2,
   return value;
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * \brief Do a 3D linear interpolation between vertices of a rectangular prism.
+ *
+ * Algorithm from wikipedia's Trilinear Interpolation article, hat tip to E.
+ * Norris for the reference.
+ *
+ * \param[in] x0   lower x coordinate of lattice
+ * \param[in] x1   upper x coordinate of lattic
+ * \param[in] y0   lower y coordinate of lattice
+ * \param[in] y1   upper y coordinate of lattice
+ * \param[in] z0   lower z coordinate of lattice
+ * \param[in] z1   upper z coordinate of lattic
+ * \param[in] f000 function at (x0,y0,z0)
+ * \param[in] f100 function at (x1,y0,z0)
+ * \param[in] f001 function at (x0,y0,z1)
+ * \param[in] f101 function at (x1,y0,z1)
+ * \param[in] f010 function at (x0,y1,z0)
+ * \param[in] f110 function at (x1,y1,z0)
+ * \param[in] f011 function at (x0,y1,z1)
+ * \param[in] f111 function at (x1,y1,z1)
+ * \param[in] x    x coordinate of interpolation point
+ * \param[in] y    y coordinate of interpolation point
+ * \param[in] z    z coordinate of interpolation point
+ * \return The function value linearly interpolated to (x,y,z)
+ */
+inline double
+linear_interpolate_3(double const x0, double const x1, double const y0,
+                     double const y1, double const z0, double const z1,
+                     double const f000, double const f100, double const f001,
+                     double const f101, double const f010, double const f110,
+                     double const f011, double const f111, double const x,
+                     double const y, double const z) {
```

Comment:
As we discussed before, I really dislike this argument list as it will be error prone (17 doubles).  Consider reworking this algorithm to use `std::array<double,2>` for (x,y) pairs and maybe a 3x3 mdspan for f999.  I would like input from @along and @RyanWollaeger on ideas to make this interface less error prone.

Reply:
It seems the argument list could be made into three arrays: one size 6 for the lattice coordinates, one size 8 for the function, and one size 3 for the point.  Maybe the data array could be a `std::map` instead, then one could have, e.g., `data["f101"]` to access that function value.

Reply:
I think arrays in the argument would be a good path forward, though I worry about performance even with `std::unordered_map`; this function may live inside heavy compute loops. From my discussion with @KineticTheory this morning, I think for now I will just move this code into the cdi_CPEloss library and leave the function signature as is -- it's not clear that there will be enough other uses for this code for it to be in ds++.

Reply:
Ah - that makes sense.  One could also define an enumeration, e.g., F000 = 0, F001 = 1, then access array values with the enumeration.